tests/test_hydro_platinum.py

```diff
@@ -34,182 +46,200 @@ def read(self, length):
+    dev = h115iPlatinumDevice
+    dev.set_speed_profile(channel='fan', profile=iter([(20, 0), (55, 100)]))
+    dev.set_speed_profile(channel='fan1', profile=iter([(30, 20), (50, 80)]))
+
+    assert dev.device.sent[-1].data[0x0b] == 0x0
+    assert dev.device.sent[-1].data[0x1d] == 7
+    assert dev.device.sent[-1].data[0x1e:0x2c] == [30, 51, 50, 204] + 5 * [60, 255]
+    assert dev.device.sent[-1].data[0x11] == 0x0
+    assert dev.device.sent[-1].data[0x2c:0x3a] == [20, 0, 55, 255] + 5 * [60, 255]
+
+    with pytest.raises(ValueError):
+        dev.set_speed_profile('invalid', [])
+
+    with pytest.raises(ValueError):
+        dev.set_speed_profile('fan', zip(range(10), range(10)))
+
+def test_h115i_platinum_device_address_leds(h115iPlatinumDevice):
+    dev = h115iPlatinumDevice
+    colors = [[i + 3, i + 2, i + 1] for i in range(0, 24 * 3, 3)]
+    encoded = list(range(1, 24 * 3 + 1))
+    dev.set_color(channel='led', mode='super-fixed', colors=iter(colors))
+
+    assert len(dev.device.sent) == 5 # 3 for enable, 2 for off
+    assert dev.device.sent[0].data[1] & 0b111 == 0b001
+    assert dev.device.sent[1].data[1] & 0b111 == 0b010
+    assert dev.device.sent[2].data[1] & 0b111 == 0b011
+    assert dev.device.sent[3].data[1] & 0b111 == 0b100
+    assert dev.device.sent[3].data[2:62] == encoded[:60]
+    assert dev.device.sent[4].data[1] & 0b111 == 0b101
+    assert dev.device.sent[4].data[2:14] == encoded[60:]
+
+def test_h115i_platinum_device_synchronize(h115iPlatinumDevice):
+    dev = h115iPlatinumDevice
+    colors = [[3, 2, 1]]
+    encoded = [1, 2, 3] * 24
+
+    dev.set_color(channel='led', mode='fixed', colors=iter(colors))
+
+    assert len(dev.device.sent) == 5 # 3 for enable, 2 for off
+
+    assert dev.device.sent[0].data[1] & 0b111 == 0b001
+    assert dev.device.sent[1].data[1] & 0b111 == 0b010
+    assert dev.device.sent[2].data[1] & 0b111 == 0b011
+    assert dev.device.sent[3].data[1] & 0b111 == 0b100
+
+    assert dev.device.sent[3].data[2:62] == encoded[:60]
+    assert dev.device.sent[4].data[1] & 0b111 == 0b101
+    assert dev.device.sent[4].data[2:14] == encoded[60:]
+
+def test_h115i_platinum_device_leds_off(h115iPlatinumDevice):
+    dev = h115iPlatinumDevice
+    dev.set_color(channel='led', mode='off', colors=iter([]))
+    assert len(dev.device.sent) == 5 # 3 for enable, 2 for off
+
+    for _, data in dev.device.sent[3:5]:
+        assert data[2:62] == [0] * 60
+
+def test_h115i_platinum_device_invalid_color_modes(h115iPlatinumDevice):
+    dev = h115iPlatinumDevice
+
+    with pytest.raises(ValueError):
+        dev.set_color('led', 'invalid', [])
+
+    with pytest.raises(ValueError):
+        dev.set_color('invalid', 'off', [])
+
+    assert len(dev.device.sent) == 0
+
+def test_h115i_platinum_device_short_enough_storage_path(h115iPlatinumDevice):
+    dev = h115iPlatinumDevice
+    #this test should not be here, the storage backed is tested seperately
```

Comment:
The reason why this was tested here is that it depends on what IDs are used to initialize the RuntimeStorage, which is driver dependent.  Also, this was a regression test, so if possible it should be restored.

Reply:
okay, Ill add that test back using the real storage driver and not the mocked one.